### PR TITLE
Support for half-bounded range operators

### PR DESCRIPTION
**Pull request by beowulfenator**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes

I've done some minor cleanup and added tests. Also I've reverted this commit: https://github.com/yiisoft/yii2-elasticsearch/pull/76/commits/5f51c73c296254394f95c2c70da21d9d073f117a.

If we do need to modify cURL code, let's do it in another PR.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ Yii Framework 2 elasticsearch extension Change Log
 - Bug: Updated debug panel classes to be consistent with yii 2.0.7 (beowulfenator)
 - Enh #15: Special data provider `yii\elasticsearch\ActiveDataProvider` created (klimov-paul)
 - Enh #60: Minor updates to guide (devypt, beowulfenator)
+- Enh: Support for "gt", ">", "gte", ">=", "lt", "<", "lte", "<=" operators in query (i-lie, beowulfenator)
```
Comment (samdark) [inline CHANGELOG.md]: Add `#83`

File: QueryBuilder.php [hunk 2]
```diff
@@ -340,6 +348,44 @@ private function buildInCondition($operator, $operands)
...
+            $range_operator = 'lt';
+        }
+
+        if (null === $range_operator) {
+            throw new InvalidParamException("Operator '$operator' is not implemented.");
+        }
+
+        $filter = [
+            'range' => [
+                "$column" => [
```
Comment (SilverFire) [inline QueryBuilder.php]: may be `$column` (without `"`)

File: QueryBuilder.php [hunk 3]
```diff
@@ -340,6 +348,44 @@ private function buildInCondition($operator, $operands)
...
+            $range_operator = 'gte';
+        } else if (in_array($operator, ['lte', '<='])) {
+            $range_operator = 'lte';
+        } else if (in_array($operator, ['gt', '>'])) {
+            $range_operator = 'gt';
+        } else if (in_array($operator, ['lt', '<'])) {
+            $range_operator = 'lt';
+        }
+
+        if (null === $range_operator) {
```
Comment (SilverFire) [inline QueryBuilder.php]: We prefer to have variable from the left of operator

File: QueryBuilder.php [hunk 4]
```diff
@@ -340,6 +348,44 @@ private function buildInCondition($operator, $operands)
...
     }
 
+    private function buildHalfBoundedRangeCondition($operator, $operands)
+    {
+        if (!isset($operands[0], $operands[1])) {
+            throw new InvalidParamException("Operator '$operator' requires two operands.");
+        }
+
+        list($column, $value) = $operands;
+        if ($column == '_id') {
```
Comment (SilverFire) [inline QueryBuilder.php]: `===`

File: QueryBuilder.php [hunk 5]
```diff
@@ -340,6 +348,44 @@ private function buildInCondition($operator, $operands)
         return $filter;
     }
 
+    private function buildHalfBoundedRangeCondition($operator, $operands)
```
Comment (SilverFire) [inline QueryBuilder.php]: How about a PHPDoc?

**Comment by beowulfenator** [comment]
Better?

File: QueryBuilder.php (hunk 4 above)
Comment (SilverFire) [inline QueryBuilder.php]: How about this?

**Pull request merged**